src/Riok.Mapperly/Symbols/MapperMethodReference.cs

```diff
@@ -0,0 +1,5 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace Riok.Mapperly.Symbols;
+
+public record MapperMethodReference(IMethodSymbol Method, INamedTypeSymbol? TargetType, ISymbol? TargetMember);
```

Comment:
Why this intermediate step and not resolving `MethodProvidedSourceValue` directly?

Reply:
I kept the original structure of the program, which kept the resolution and creation separate. I can merge it into one, if you wish.

Reply:
I merged this

Reply:
Merge is done